src/UglyToad.PdfPig/DocumentLayoutAnalysis/NNWordExtractor.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UglyToad.PdfPig.Content;
+using UglyToad.PdfPig.Geometry;
+using UglyToad.PdfPig.Util;
+
+namespace UglyToad.PdfPig.DocumentLayoutAnalysis
+{
+    /// <summary>
+    /// Nearest Neighbour Word Extractor, using the <see cref="Distances.Manhattan"/> distance.
+    /// This implementation leverages bounding boxes.
+    /// </summary>
+    public class NNWordExtractor : IWordExtractor
```

Comment:
On naming here (I'm annoying sorry!) I think the full name is going to help public api discovery:

```suggestion
    public class NearestNeighbourWordExtractor : IWordExtractor
```